webapp/channels/src/components/admin_console/localization/localization.tsx

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import type {LocalizationSettings} from '@mattermost/types/config';
+
+import BooleanSetting from 'components/admin_console/boolean_setting';
+import DropdownSetting from 'components/admin_console/dropdown_setting';
+import MultiSelectSetting from 'components/admin_console/multiselect_settings';
+import {
+    AdminSection,
+    SectionContent,
+    SectionHeader,
+} from 'components/admin_console/system_properties/controls';
+import ExternalLink from 'components/external_link';
+
+import * as I18n from 'i18n/i18n.jsx';
+
+import type {SystemConsoleCustomSettingsComponentProps} from '../schema_admin_settings';
+import './localization.scss';
+
+const locales = I18n.getAllLanguages();
+
+export default function Localization(props: SystemConsoleCustomSettingsComponentProps) {
+    const [localizationSettings, setLocalizationSettings] = useState<LocalizationSettings>(props.value as LocalizationSettings);
+
+    const handleChange = useCallback((id: string, value: any) => {
+        const updatedSettings = {
+            ...localizationSettings,
+            [id]: value,
+        };
+        setLocalizationSettings(updatedSettings);
+        props.onChange(props.id, updatedSettings);
+    }, [props, localizationSettings]);
+
+    const availableLanguages = useMemo(() => {
+        const values: Array<{value: string; text: string; order: number}> = [];
+        for (const l of Object.values(locales)) {
+            values.push({value: l.value, text: l.name, order: l.order});
+        }
+        values.sort((a, b) => a.order - b.order);
+        return values;
+    }, []);
+
+    return (
+        <AdminSection style={{marginTop: '0'}}>
+            <SectionHeader>
+                <hgroup>
+                    <h1 className='localization-section-title'>
+                        <FormattedMessage
+                            id='admin.site.localization.languages.title'
+                            defaultMessage='Languages'
+                        />
+                    </h1>
+                    <h5 className='localization-section-description'>
+                        <FormattedMessage
+                            id='admin.site.localization.languages.description'
+                            defaultMessage='Choose which languages should be the defaults'
+                        />
+                    </h5>
+                </hgroup>
+            </SectionHeader>
+
+            <SectionContent>
+                <DropdownSetting
+                    key={props.id + '_language_' + props.id + '.DefaultServerLocale'}
```

Comment:
Why adds keys to all of these? They also looks super odd with repeating the ID in them twice

Reply:
removed the key on all of them